### PR TITLE
Add verify address button to Receive modal

### PR DESCRIPTION
**Pull request by brusherru**
It is requested by Ledger company to add a "verify" button on "Receive" modal, that asks User to verify the address on his Ledger device. So now it is implemented.

It may require design improvement or rephrasing.

image
image
image

And there is also possible another case with red text, saying: "The address is incorrect, probably you are using another Ledger device." :)

### CODE REVIEW

File: src/components/Receive.tsx
Comment (monikasmolarek) [inline src/components/Receive.tsx:144]:
```suggestion
                Please review the address on the Ledger device and approve if it 
                matches the address above.
```

**Review by monikasmolarek — CHANGES_REQUESTED** [review]
Once the address check is rejected on the ledger, it gets automatically the modal claiming (incorrectly) the ledger is disconnected. 

+ the error messages were messy🤔

https://github.com/user-attachments/assets/fab38f2a-478e-4ff5-ad11-de5084af355a

**Comment by brusherru** [comment]
@monikasmolarek thanks!
Fixed. There are two new messages for the cases:
1. When the address is rejected on the device, it does not require to reconnect anymore, but displays:
   image
    
2. When the device is locked or not connected there will be another message (in case User closes the "connect device" popup). Example:
    image
    image

**Review by monikasmolarek — COMMENTED** [review]
@brusherru This is indeed fixed, thanks ! 
but please double-check the case when you start the address verification, you don't do anything on the ledger, so the request there is waiting for your approve/reject, but in the Wallet you click OK or close the modal etc. 
When the tx/action is pending on the device, the Wallet gets really confused about the device connection. 

https://github.com/user-attachments/assets/e45e9763-9109-40df-93e5-fbdcc5a10720

My steps in this rec: 
1. opened receive, clicked verify -> checked the device but didn’t approve or reject, left pending
2. clicked OK on the modal
3. opened receive, clicked verify again-> the reconnect modal appeared
4. took device, approved the request -> it got approved "in the background" below the reconnect modal
5. the wallet got confused about the connection

**Comment by brusherru** [comment]
> When the tx/action is pending on the device, the Wallet gets really confused about the device connection.

The problem here that we cannot cancel the operation on the ledger device if you close the window.
And at the same time, Ledger works in a synchronous way, which means if you run another operation while the previous one was not finished — it causes transport error, which also causes disconnecting from the device.

The only thing I can do here is to disable closing the popup once "Verify" button was clicked.
So I tweaked it, and hopefully it will work well :D

image


However, if you reload the page and click "Verify" once again — we don't have any chance to handle it properly in the app.

**Comment by brusherru** [comment]
Ok, I'm pretty sure it is fine, so I'm merging it.
If we find out any problems — I'll fix it :)

**Pull request merged**
